splitgraph/hooks/data_source/base.py

```diff
@@ -302,3 +309,113 @@ def prepare_new_image(
         base_image = None
         repository.images.add(parent_id=None, image=new_image_hash, comment=comment)
     return base_image, new_image_hash
+
+
+def init_write_overlay(
```

Comment:
I think this function (and the three constants above) should be moved into a separate module in `splitgraph.core` (overlay.py?). It doesn't seem like it depends on anything in `data_source` (apart from maybe being kind of similar to mounting?) and most users of these symbols are in `splitgraph.core`. The other issue is that anything that wants to use this `init_write_overlay` code or reference the constants will now have to pull in a lot of ingestion / data source specific code:

```bash
$ ./dc-pants.sh dependencies --transitive src/py/splitgraph/splitgraph/hooks/data_source/base.py | grep -e ingestion -e resources -e data_source
src/py/splitgraph/splitgraph/hooks/data_source/__init__.py
src/py/splitgraph/splitgraph/hooks/data_source/base.py
src/py/splitgraph/splitgraph/hooks/data_source/fdw.py
src/py/splitgraph/splitgraph/hooks/data_source/utils.py
src/py/splitgraph/splitgraph/ingestion/airbyte/data_source.py
src/py/splitgraph/splitgraph/ingestion/airbyte/docker_utils.py
src/py/splitgraph/splitgraph/ingestion/airbyte/models.py
src/py/splitgraph/splitgraph/ingestion/airbyte/utils.py
src/py/splitgraph/splitgraph/ingestion/common.py
src/py/splitgraph/splitgraph/ingestion/dbt/utils.py
src/py/splitgraph/splitgraph/ingestion/singer/common.py
src/py/splitgraph/splitgraph/resources/icons/__init__.py
src/py/splitgraph/splitgraph/resources/icons/athena.svg
src/py/splitgraph/splitgraph/resources/icons/bigquery.svg
src/py/splitgraph/splitgraph/resources/icons/csv.svg
src/py/splitgraph/splitgraph/resources/icons/dbt.svg
src/py/splitgraph/splitgraph/resources/icons/elasticsearch.svg
src/py/splitgraph/splitgraph/resources/icons/mongodb.svg
src/py/splitgraph/splitgraph/resources/icons/mysql.svg
src/py/splitgraph/splitgraph/resources/icons/postgresql.svg
src/py/splitgraph/splitgraph/resources/icons/snowflake.svg
src/py/splitgraph/splitgraph/resources/icons/socrata.svg
src/py/splitgraph/splitgraph/resources/splitgraph_meta/__init__.py
src/py/splitgraph/splitgraph/resources/splitgraph_meta/splitgraph_meta--0.0.1--0.0.2.sql
src/py/splitgraph/splitgraph/resources/splitgraph_meta/splitgraph_meta--0.0.1.sql
src/py/splitgraph/splitgraph/resources/splitgraph_meta/splitgraph_meta--0.0.2--0.0.3.sql
src/py/splitgraph/splitgraph/resources/splitgraph_meta/splitgraph_meta--0.0.3--0.0.4.sql
src/py/splitgraph/splitgraph/resources/static/__init__.py
src/py/splitgraph/splitgraph/resources/static/audit_trigger.sql
src/py/splitgraph/splitgraph/resources/static/cstore.sql
src/py/splitgraph/splitgraph/resources/static/splitgraph_api.sql
```

meaning that if we ever use Pants' dependency inference to choose whether or not to run unit tests, we'll falsely end up running more irrelevant tests if changing something like `ingestion/airbyte/data_source.py` than necessary.

Reply:
Makes sense, moving to a separate file.

Reply:
Also one terminology question—what do you think about shadow/source as opposed to upper/lower naming of the overlay component tables?

Reply:
No particular opinion TBH. I borrowed the naming from OverlayFS/Docker (https://www.kernel.org/doc/html/latest/filesystems/overlayfs.html#upper-and-lower) but shadow/source sounds more intuitive without having to know the context. Feel free to change it if you like it more!

Reply:
Nice, I didn't realize that. I'd leave it as is then, since the reference you provide should be handy when describing how this works in docs/blogs/HNs.